src/mono/mono/metadata/assembly-internals.h

```diff
@@ -141,6 +141,30 @@ mono_assembly_get_image_internal (MonoAssembly *assembly);
 void
 mono_set_assemblies_path_direct (char **path);
 
+typedef enum {
+	BUNDLED_BLOB,
+	BUNDLED_ASSEMBLY,
+	BUNDLED_SATELLITE_ASSEMBLY,
+	BUNDLED_BINARY,
```

Comment:
Is there a known set of resource types we expect to need to bundle? I was using BLOB as a default to handle any resource that didn't fit assembly (.dll), satellite assembly (.resources.dll), or binary (.bin). For example if this task was used to also bundle a plain text file or something.
As is, the enum would help bucket the different assemblies so they can be post processed accordingly. Right now the only post processing that is resource specific are the apis that register assemblies and satellite assemblies, so it would be good to be able to just grab those types. Do we anticipate some sort of registering for binaries or plain text or some other type of resource? If not then we can just bundle them as BINARY as the default.

Reply:
or to make it more inclusive, maybe just call it data as we have in the API, then it can be binary/text etc, so BUNDLED_ASSEMBLY, BUNDLED_SATELLITE_ASSEMBLY, BUNDLED_DATA ?

Reply:
Will do, and if we need to do more post processing of specific resources we can add more types on a need to basis.

Reply:
How is the type used, which logic needs to read it ?

Reply:
One of the objectives for bundling with the self-contained library builder was to bundle `runtimeconfig.bin`, so that will be a `MONO_BUNDLED_DATA` type as well as probably all of the timezone data as well.

The `BundledResourceType` is used to differentiate between different `BundledResource`s for purposes of post processing like `mono_register_bundled_assemblies` and `mono_register_bundled_satellite_assemblies` or even future APIs of "get me all bundled resources of this type" etc (if theres a need for that).